python/tvm/analysis/__init__.py

```diff
@@ -0,0 +1,296 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""High level analysis functions"""
```

Comment:
Would be good to think about the namespacing choices. 

This is a mixture of profiling that is also related to TIR features. In before analysis under `tir/analysis` or `relay/analysis` were about IR choices.

How about `contrib/roofline` if we aimed at roofline analysis?

Some of the features fit into tir/profile?

Reply:
I wasn't sure about the correct namespace either. It doesn't really fit in `tir` or `relay` because it is about high level features, not anything specific to tir or relay. It doesn't fit in `runtime` because it requires auto scheduler, so I can't put it in `runtime/profiling`. Also, it is not really profiling, it is an analysis.

I don't think `contrib/roofline` is a good match because contrib is poorly defined and adds no extra information. `roofline` doesn't make sense because there are more functions than just the roofline model (estimated flops and bandwidth).

The reason I choose `analysis` is because it seemed better than all the choices listed above. Do you have a reason not to use `analysis`?

Reply:
The main reason was mentioned above: `analysis` as reserved for IR based analysis itself(in the case of tir/relay analysis). So there might be confusions in here 

In this case `profile_analysis` might be a good distinction part(it was not the best name that i can come up with), happy to discuss other name choices as well

Reply:
Coming back after more deliberations. A new namespace is a main architectural choice as well as API choice itself that would benefit from broader [deliberation and discussions](https://tvm.apache.org/docs/contribute/code_review.html#deliberate-on-api-and-data-structures). Especially when we are not too sure about the choice of the name.

In the meantime, the particular module started with LLVM cpu target and is more like a standalone tooling that to be continuously improved and used. Our previous convention would start with such self-contained tooling in contrib, e.g.
`contrib/popen_pool`, which can use less deliberations architectural-wise and unblock the PR. In the meantime, we can open a discuss thread to see if others's input about possible ways to group the things as they matures.

Please let me know if that makes sense.